luwak/src/test/java/uk/co/flax/luwak/TestMonitorStandardQueryParser.java

```diff
@@ -0,0 +1,318 @@
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+public class TestMonitorStandardQueryParser {
+
+    static final String TEXTFIELD = "TEXTFIELD";
+
+    static final Analyzer ANALYZER = new WhitespaceAnalyzer();
+
+    private Monitor monitor;
+    private StandardLuceneQueryParser parser;
+    @Before
+    public void setUp() throws IOException {
+        parser = new StandardLuceneQueryParser(TEXTFIELD, ANALYZER);
+        monitor = new Monitor(parser, new MatchAllPresearcher());
+    }
+
+    @Test
+    public void singleTermQueryMatchesSingleDocument() throws IOException, UpdateException {
+
+        String document = "This is a test document";
+
+        DocumentBatch batch = DocumentBatch.of(InputDocument.builder("doc1")
+                .addField(TEXTFIELD, document, ANALYZER)
+                .build());
+
+        monitor.update(new MonitorQuery("query1", "test"));
+
+        assertThat(monitor.match(batch, SimpleMatcher.FACTORY))
+                .matchesDoc("doc1")
+                .hasMatchCount("doc1", 1)
+                .matchesQuery("query1", "doc1");
+
+    }
+
+    @Test
+    public void matchStatisticsAreReported() throws IOException, UpdateException {
+        String document = "This is a test document";
+        DocumentBatch batch = DocumentBatch.of(InputDocument.builder("doc1")
+                .addField(TEXTFIELD, document, ANALYZER)
+                .build());
+
+        monitor.update(new MonitorQuery("query1", "test"));
+
+        Matches<QueryMatch> matches = monitor.match(batch, SimpleMatcher.FACTORY);
+        Assertions.assertThat(matches.getQueriesRun()).isEqualTo(1);
+        Assertions.assertThat(matches.getQueryBuildTime()).isGreaterThan(-1);
+        Assertions.assertThat(matches.getSearchTime()).isGreaterThan(-1);
+    }
+
+
+    @Test
+    public void matchIntPointfields() throws IOException,UpdateException,Exception {
+
+        parser.getPointsConfig().put("age",new PointsConfig(NumberFormat.getIntegerInstance(Locale.ROOT), Integer.class));
+
+
+        InputDocument doc = InputDocument.builder("doc1")
+                .addField(new IntPoint("age",1))
+                .addField(new IntPoint("somethingelse",2))
+                .build();
+
+        MonitorQuery mq  = new MonitorQuery("query1","(age:1 somethingelse:2)");
+        monitor.update(mq);
+        Matches<?> matches = monitor.match(doc, SimpleMatcher.FACTORY);
+        assertThat(matches)
+        .hasMatchCount("doc1", 1);
+
+        mq  = new MonitorQuery("query1","age:1");
+        monitor.update(mq);
+        matches = monitor.match(doc, SimpleMatcher.FACTORY);
+        assertThat(matches)
+        .hasMatchCount("doc1", 1);
+    }
+
+    @Test
```

Comment:
I think we can remove everything below here?  It's just duplicating the tests in TestMonitor, right?